components/org.wso2.bfsi.consent.management.extensions/src/main/java/org/wso2/bfsi/consent/management/extensions/common/idempotency/IdempotencyValidationUtils.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.consent.management.extensions.common.idempotency;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.bfsi.consent.management.common.config.ConsentManagementConfigParser;
+import org.wso2.bfsi.consent.management.common.exceptions.ConsentManagementException;
+import org.wso2.bfsi.consent.management.extensions.internal.ConsentExtensionsDataHolder;
+import org.wso2.bfsi.consent.management.service.ConsentCoreService;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class to hold idempotency validation utils.
+ */
+public class IdempotencyValidationUtils {
+
+    private static final Log log = LogFactory.getLog(IdempotencyValidationUtils.class);
+    private static final ConsentCoreService consentCoreService = ConsentExtensionsDataHolder.getInstance()
+            .getConsentCoreService();
+
+    /**
+     * Method to retrieve the consent ids that have the idempotency key name and value as attribute.
+     *
+     * @param idempotencyKeyName     Idempotency Key Name
+     * @param idempotencyKeyValue    Idempotency Key Value
+     * @return   List of consent ids if available, else an empty list will be returned
+     */
+    static List<String> getConsentIdsFromIdempotencyKey(String idempotencyKeyName,
+                                                        String idempotencyKeyValue) {
+        try {
+            return consentCoreService.getConsentIdByConsentAttributeNameAndValue(
+                    idempotencyKeyName, idempotencyKeyValue);
+        } catch (ConsentManagementException e) {
+            log.debug("No consent ids found for the idempotency key value");
```

Comment:
Should this be an error log?

Reply:
No this means the idempotency key has not been replayed